src/3rd_party-static/message_broker/src/lib_messagebroker/CMessageBroker.cpp

```diff
@@ -444,26 +444,37 @@ void CMessageBroker::onMessageReceived(int fd, std::string& aJSONData, bool tryH
   while (! aJSONData.empty()) {
     Json::Value root;
     if ((! p->m_reader.parse(aJSONData, root)) || root.isNull()) {
-      DBG_MSG_ERROR(("Unable to parse JSON!"));
+      DBG_MSG_ERROR(("Unable to parse JSON!\n"));
       if (! tryHard) {
         return;
       }
       uint8_t first_byte = static_cast<uint8_t>(aJSONData[0]);
       if ((first_byte <= 0x08) || ((first_byte >= 0x80) && (first_byte <= 0x88))) {
-        DBG_MSG((" There is an unparsed websocket header probably.\n"));
+        DBG_MSG(("There is an unparsed websocket header probably.\n"));
         /* Websocket headers can have FIN flag set in the first byte (0x80).
          * Then there are 3 zero bits and 4 bits for opcode (from 0x00 to 0x0A).
          * But actually we don't use opcodes above 0x08.
          * Use this fact to distinguish websocket header from payload text data.
          * It can be a coincidence of course, but we have to give it a try. */
         return;
       } else if ('{' == aJSONData[0]) {
-        DBG_MSG_ERROR((" Incomplete JSON object probably.\n"));
-        return;
+        const bool is_object = true;
+        const size_t next_object_pos =
+            p->jumpOverJSONObjectOrArray(is_object, aJSONData);
+
+        if (next_object_pos != std::string::npos) {
+          DBG_MSG_ERROR(("Invalid JSON object probably. Skipping.\n"));
+          aJSONData.erase(0, next_object_pos);
+          DBG_MSG(("Buffer after cut is: '%s'\n", aJSONData.c_str()));
+          continue;
+        } else {
```

Comment:
@AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/1590/commits/2e33280a1c6cb077fa5a301f967c859318501e24